### DISCUSSION TITLE
boto3 version compatibality

### DISCUSSION THREAD

**Discussion by itsanand** [aio-libs/aiobotocore#982] [Q&A]
Why aiobotcore is not compatible with latest version of boto3

**Comment by baggiponte** [comment]
Hello, I was interested in this too. While `botocore` is pinned to `'botocore>=1.29.76,<1.29.77'`, why is `boto3` bound to `'boto3>=1.26.76,<1.26.77'`, three minor versions behind?

**Reply by baggiponte** [reply]
Hi @thehesiod, hate to ping you directly. I looked at the issues and labels and could not see anything related to this. Do you think aiobotocore should support a more recent boto3 release? Does not have to be up to date - perhaps just support the same botocore minor version. I so, I would like to help out

**Comment by thehesiod** [comment]
oh no worry at all!  btw how do you ping someone directly on github?  I'd
like to let other people know how to do this.

yes! we're always open to PRs to bump aiobotocore.  I haven't had much time
recently to do this myself.  I have instructions here:
https://github.com/aio-libs/aiobotocore/blob/master/CONTRIBUTING.rst#how-to-upgrade-botocore.
Here's a recent PR that was done to bump it:
https://github.com/aio-libs/aiobotocore/pull/999

**Reply by baggiponte** [reply]
Doesn't seem evil, I will have a look at that (ideally next week). How about boto3?

**Reply by baggiponte** [reply]
> btw how do you ping someone directly on github?  I'd like to let other people know how to do this.

I just @ your username - I guess it worked!

**Reply by thehesiod** [reply]
oh! lol, this was just a github email notification

**Reply by thehesiod** [reply]
boto3/awscli are all version locked to match our botocore version lock

**Reply by baggiponte** [reply]
> boto3/awscli are all version locked to match our botocore version lock

Is it possible to relax the constraint?

**Reply by thehesiod** [reply]
yes, but then we couldn't guarantee the library would work.  you can always yourself install a newer version of boto3/awscli and ignore the pep warnings/errors.

**Reply by baggiponte** [reply]
> yes, but then we couldn't guarantee the library would work.

Do you mean it is only guaranteed to work for the locked version?

> you can always yourself install a newer version of boto3/awscli and ignore the pep warnings/errors.

I could do so by cloning aiobotocore and removing the constrant?

**Reply by thehesiod** [reply]
correct, because boto3/awscli are tied to botocore versions, and we're tied to a specific botocore version. I wouldn't do that, I'd just re-install boto3/awscli after installing aiobotocore

**Reply by baggiponte** [reply]
> I'd just re-install boto3/awscli after installing aiobotocore

I tried this but it automatically installs the same boto3 version as the locked one

> correct, because boto3/awscli are tied to botocore versions, and we're tied to a specific botocore version

That makes sense, but I am left wondering why aiobotocore pins boto3 to [v 1.26](https://github.com/aio-libs/aiobotocore/blob/72b8dd5d7d4ef2f1a49a0ae0c37b47e5280e2070/setup.py#L18) when the pinned botocore is 1.29. Does this mean boto3 could be bumped to 1.29?

**Reply by thehesiod** [reply]
because of this: https://github.com/boto/boto3/blob/1.26.76/setup.py#L16

**Reply by baggiponte** [reply]
oh shoot, my bad! I believed boto3 was on the same minor as botocore, but it is three versions behind. lol. thanks! When I have time I will look at bumping the minor on botocore!
